src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/subquery/rewriter/InRewriter.java

```diff
@@ -0,0 +1,117 @@
+ * For example,
+ * SELECT * FROM A WHERE a IN (SELECT b FROM B) and c > 10 should be rewritten to
+ * SELECT A.* FROM A JOIN B ON A.a = B.b WHERE c > 10 and B.b IS NOT NULL.
+ */
+public class InRewriter implements Rewriter {
+    private final SQLInSubQueryExpr inExpr;
+    private final BlackBoard bb;
+    private final MySqlSelectQueryBlock queryBlock;
+
+    public InRewriter(SQLInSubQueryExpr inExpr, BlackBoard bb) {
+        this.inExpr = inExpr;
+        this.bb = bb;
+        this.queryBlock = (MySqlSelectQueryBlock) inExpr.getSubQuery().getQuery();
+    }
+
+    //
+    @Override
+    public boolean canRewrite() {
+        return !inExpr.isNot();
+    }
+
+    /**
+     * Build Where clause from input query.
+     * <p>
+     * With the input query.
+     *         Query
+     *      /    |     \
+     *  SELECT  FROM  WHERE
+     *  |        |    /   |  \
+     *  *         A  c>10 AND INSubquery
+     *                         /    \
+     *                        a    Query
+     *                        /      \
+     *                           SELECT FROM
+     *                             |     |
+     *                             b     B
+     * <p>
+     *
+     */
+    @Override
+    public void rewrite() {
+        SQLTableSource from = queryBlock.getFrom();
+        addJoinTable(from);
+
+        SQLExpr where = queryBlock.getWhere();
+        if (null == where) {
+            bb.addWhere(generateNullOp());
+        } else if (where instanceof SQLBinaryOpExpr) {
+            bb.addWhere(and(generateNullOp(), (SQLBinaryOpExpr) where));
+        } else {
+            throw new IllegalStateException("unsupported where class type " + where.getClass());
+        }
+    }
+
+    /**
+     * Build the Null check expression. For example,
+     * SELECT * FROM A WHERE a IN (SELECT b FROM B), should return B.b IS NOT NULL
+     */
+    private SQLBinaryOpExpr generateNullOp() {
+        SQLBinaryOpExpr binaryOpExpr = new SQLBinaryOpExpr();
+        binaryOpExpr.setLeft(fetchJoinExpr());
+        binaryOpExpr.setRight(new SQLNullExpr());
+        binaryOpExpr.setOperator(SQLBinaryOperator.IsNot);
+
+        return binaryOpExpr;
+    }
+
+    /**
+     * Add the {@link SQLTableSource} with {@link JoinType} and {@link SQLBinaryOpExpr} to the {@link BlackBoard}.
+     */
+    private void addJoinTable(SQLTableSource right) {
+        SQLBinaryOpExpr binaryOpExpr = new SQLBinaryOpExpr(inExpr.getExpr(),
+                SQLBinaryOperator.Equality,
+                fetchJoinExpr());
+        bb.addJoin(right, JoinType.JOIN, binaryOpExpr);
+    }
+
+    private SQLExpr fetchJoinExpr() {
+        if (queryBlock.getSelectList().size() > 1) {
+            throw new IllegalStateException("Unsupported subquery with multiple select " + queryBlock.getSelectList());
+        }
+        return queryBlock.getSelectList().get(0).getExpr();
+    }
+
+    private SQLBinaryOpExpr and(SQLBinaryOpExpr left, SQLBinaryOpExpr right) {
```

Comment:
could move to a `default` method in `Rewriter` interface since it's shared by different implementations.

Reply:
Done.